lang/java/protobuf/src/main/java/org/apache/avro/protobuf/ProtobufData.java

```diff
@@ -359,4 +366,16 @@ private JsonNode getDefault(FieldDescriptor f) {
 
   }
 
+  private Map<Descriptor, Schema> schemaConversions = new HashMap<>();
+
+  /**
+   * Set proto descriptor -> avro schema conversion to handle logical type
+   *
+   * @param descriptor message descriptor in protobuf
+   * @param avroSchema avro schema might have logical type
+   */
+  public void addSchemaConversion(Descriptor descriptor, Schema avroSchema) {
```

Comment:
I don't really get the purpose of this new method. The parent ProtobufData class already provides a way to register conversions via addLogicalTypeConversion, why is an additional addSchemaConversion needed? Actually Protobuf has similarities with ReflectData: Avro tries to infer the schema based on the type of the field, in this case from Timestamp. Using addLogicalTypeConversion at client side, and getConversionByClass in ProtobufData would do the same without adding an additional method, clients can register conversion like they do for non-Protobuf Avro classes like GenericData, ReflectData. What do you think about this approach?

Reply:
I was just missing `getConversionByClass()` ... we can replace this with the method I think :)

Reply:
I did it https://github.com/apache/avro/pull/482/commits/01cbf73a3cb6c25b877daa2de7888c46e89a22ad , it supports to resolve conversions from descriptor without additional method calls like `addSchemaConversion()`. What do you think?

Reply:
@syucream I like this approach more than adding an extra public method! I left one comment, otherwise fine, if all test pass and serves your need, then I prefer this. Could you please address my other comments? I'd be happy to merge this.